### PR TITLE
use MonadIO to generalise the library a bit.

### PR DESCRIPTION
**Pull request by L0neGamer**

### CODE REVIEW

**Comment by L0neGamer** [comment]
A couple of parameters (such as ClientApp and Config) still use IO, but there's not much I can do about that without breaking changes/changing something upstream.

File: source/library/Wuss.hs [hunk 1]
```diff
@@ -152,20 +158,21 @@ defaultConfig = do
...
-  -> IO.IO a
+  -> m a
 runSecureClientWithConfig host port path config options headers app = do
-  context <- Connection.initConnectionContext
-  Exception.bracket
-    (Connection.connectTo context (connectionParams host port))
-    Connection.connectionClose
-    (\connection -> do
+  context <- MonadIO.liftIO Connection.initConnectionContext
+  UnliftIO.bracket
```
Comment (tfausak) [inline source/library/Wuss.hs:172]: Could this use `Control.Monad.Catch.bracket` and `Control.Monad.IO.Class.liftIO` instead? Then I wouldn't need to depend on `unliftio`, and the constraint would be `MonadIO m, MonadThrow m`.

**Review by tfausak — COMMENTED** [review]
I'd prefer to follow the lead of the `websockets` package whenever possible. It doesn't depend on `unliftio`, so I'm a little hesitant to add it here.

File: source/library/Wuss.hs (hunk 1 above)
Comment (L0neGamer) [reply source/library/Wuss.hs]: MonadMask is needed for bracket, but sure

Comment (L0neGamer) [reply source/library/Wuss.hs]: The downside being that it won't work with unliftio in the same way ofc

File: wuss.cabal [hunk 2]
```diff
@@ -36,6 +36,7 @@ common library
     , connection >= 0.3.1 && < 0.4
     , network >= 3.1.1 && < 3.2
     , websockets >= 0.12.7 && < 0.13
+    , exceptions >= 0.10
```
Comment (tfausak) [inline wuss.cabal:39]:
```suggestion
    , exceptions >= 0.10 && < 0.11
```

**Review by tfausak — APPROVED** [review]
Fantastic, thank you!

**Pull request merged**